packages/node_modules/@ciscospark/spark-core/src/lib/batcher.js

```diff
@@ -166,12 +166,15 @@ const Batcher = SparkPlugin.extend({
    * @param {SparkHttpError} reason
    * @returns {Promise<undefined>}
    */
-  handleHttpError(reason) {
+  handleHttpError(reason = `An undefined http error has occurred.`) {
     const msg = reason.message || reason.body || reason;
-    return Promise.all(reason.options.body.map((item) => this.getDeferredForRequest(item)
-      .then((defer) => {
-        defer.reject(msg);
-      })));
+    if (reason && reason.options && reason.options.body && reason.options.body.map) {
```

Comment:
use lodash.has?

Reply:
I think we might want to log something indicating that we hit the http error handler without an http error; that *should* imply that the error happened before the request was ever submitted.